lib/ecto/adapters/sql.ex

```diff
@@ -141,11 +146,17 @@ defmodule Ecto.Adapters.SQL do
   def to_sql(kind, repo, queryable) do
     adapter = repo.__adapter__
 
-    {_meta, prepared, params} =
+    plan =
       Ecto.Queryable.to_query(queryable)
       |> Ecto.Query.Planner.query(kind, repo, adapter)
-
-    {prepared, params}
+    case plan do
+      {:execute, _meta, %{statement: statement}, params} ->
```

Comment:
I think there is a bit of confusion because we have three states and execute is representing two. The states are: planned for execute, planned for prepare_execute and prepared for execute. It also doesn't help that our adapter prepare is not the database prepare.

So I'd suggest at least one change: let's rename our adapter prepare to plan.

Regarding the double meaning in execute, should we introduce three APIs? execute_planned, prepare_and_execute_planned and execute_prepared.

Btw, I can work on those changes after merged. I'd just like to know if they make sense. :)